backend/src/main/java/com/htecgroup/skynest/service/impl/RefreshTokenServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.htecgroup.skynest.service.impl;
+
+import com.htecgroup.skynest.exception.UserException;
+import com.htecgroup.skynest.exception.UserExceptionType;
+import com.htecgroup.skynest.model.dto.LoggedUserDto;
+import com.htecgroup.skynest.model.jwtObject.JwtObject;
+import com.htecgroup.skynest.service.CurrentUserService;
+import com.htecgroup.skynest.service.RefreshTokenService;
+import com.htecgroup.skynest.util.JwtUtils;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+@Log4j2
+public class RefreshTokenServiceImpl implements RefreshTokenService {
+
+  private CurrentUserService currentUserService;
+
+  @Override
+  public String refreshToken(String refreshToken) {
+    if (refreshToken != null && refreshToken.startsWith(JwtUtils.TOKEN_PREFIX)) {
```

Comment:
I don't think refreshToken can ever be null here.

Since you're using the RequestHeader annotation in the controller, if the request doesn't contain this header at all won't Spring throw an error without even getting here?

If the header is present, then refreshToken won't be null anyway.

Reply:
It depends on required flag in RequestHeader annotation. if required is true then spring will throw exception if required is false then headed will be null

Reply:
I haven't specified is it required or not, not sure what the default value is, I think is required true.

Reply:
Still this check could stay here, no need to remove it